src/CookiePhysicsComponent.h

```diff
@@ -3,6 +3,13 @@
 
 #include "PhysicsComponent.h"
 
+typedef struct {
+    int hits;
+    double launchTime;
+    glm::vec3 launchPosition;
+    glm::vec3 hitPosition;
```

Comment:
Maybe do a list of 'hits' and then calculate distance between each two consecutive points and set to total distance? Probably need to add to hits even hits that occur on houses that have already been hit previously as well (in case of multiple bounces between two houses). So some hits are just used to add to distance, some increase by 500 points (I think this is pretty much what you are saying about but it's 5am so I'm reiterating for my own sake :P )